### PR TITLE
Generator Model Training Orchestration

### PR DESCRIPTION
**Pull request by MattsonCam**
This PR expands the generator-model training workflow into a more reusable orchestration pattern and simplifies the training stack around the current reconstruction-based approach. It adds broader run configuration for optimization, checkpointing, resume behavior, and training/evaluation precision control, while also improving deterministic data handling and evaluation behavior. The result is a cleaner foundation for training current and future generator models with more consistent experiment management.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -157,11 +179,19 @@ def __call__(self, trial: optuna.trial.Trial):
         # Let Optuna choose a mini-batch size and learning rate for this trial.
         batch_size = trial.suggest_int("batch_size", 1, 8)
         lr = trial.suggest_float("lr", 1e-5, 1e-3, log=True)
```
Comment (wli51) [inline train.py]: I saw these independent optuna suggestions of lr and batch size the past several times I reviewed PRs in this repo and didn't think too much about them. Now in context of the difficulties you are encountering when getting these models to predict objects, potentially due to a low signal to noise/dynamic range problem, I think the alignment of learning rate and batch size will be important. The deep learning best practice suggests that the batch size and learning rate should be inversely related where larger batch size correspond to smaller lr. In the case of nuclear speckle data, I think this theory is relevant. The model would struggle with unorthodox parameter combinations such as batch size of 1 and a very small learning rate which can result in getting stuck in local optima one after another and prevent convergence.   
Perhaps the optuna learning parameter should be batch size and a scaling factor to lr (`a*\sqrt(k)` where `k` is batch size and `a` is the optuna searched factor). The base of lr can be set as some generic numerical value such as `1e-3` and it gets multiplied by `s\sqrt(k)` to yield the per trial lr. This would help optuna to learn and jointly optimize the two related hyperparameters.

File: README.md [hunk 2]
```diff
@@ -25,6 +33,26 @@ Within each cache root, training uses:
 
 Before samples are passed through the model, both input and target crops are normalized by their channel max intensity (dtype max, for example `255` for `uint8` and `65535` for `uint16`) in `datasets/dataset_00/utils/ImagePreProcessor.py`.
```
Comment (wli51) [inline README.md]: Are you going with the 8bit normalization + no output activation path here? I remember hearing that during stand up and I saw you setting the default output normalization. as none. Maybe the relationship between normalization method and activation should be included here?

File: README.md [hunk 3]
```diff
@@ -42,12 +70,23 @@ uv run train.py --epochs 1 --n-trials 1 --max-train-batches 2 --max-eval-batches
...
+- Epoch-end evaluation and MLflow signature inference can also run under autocast when evaluation AMP is enabled.
+- When AMP is enabled, this training path uses `bfloat16` autocast.
+
+In practice this allows training and evaluation inference behavior to be configured separately while keeping the rest of the training loop unchanged.
+
 ## Loss and Metrics
 
 - Optimization loss: L1
-- Logged metrics: L1, L2, PSNR, SSIM
+- Logged metrics: L1, L2, PSNR, SSIM, Pearson correlation
```
Comment (wli51) [inline README.md:87]: I also remember SSIM is very sensitive to numerical ranges of the data. If you switch from bitdepth max to 8bit raw normalization and still use ssim as metric, consider setting the `data_range` parameter to the empirical data min and max as opposed to the theoretical 255. The torchmetric MSSSIM has built in dynamic data range detection if you leave the parameter to `None` that would help. The provided data range max is used internally in SSIM computation to stabilize local convolution based assessment of contrast and structure and providing a mis-aligned max will cause the stabilization factor to dilute realistic change.

**Review by wli51 — APPROVED** [review]
@MattsonCam great work! This is a very difficult dataset to train on. Hopefully improvements but I wouldn't be surprised if the dynamic range still breaks everything.

File: README.md (hunk 3 above)
Comment (MattsonCam) [reply README.md:87]: This is a good suggestion that I might implement in the future because I am need to move quickly. I think this matters less right now since I am using the sigmoid, but I this may change

File: README.md (hunk 2 above)
Comment (MattsonCam) [reply README.md]: Good catch. I clarified the README to note that target normalization and output activation are intentionally split: targets are normalized to [0, 1], the model returns logits by default, and sigmoid is applied later in postprocessing when needed.

File: train.py (hunk 1 above)
Comment (MattsonCam) [reply train.py]: Thanks for the suggestion. I implement it

**Comment by MattsonCam** [comment]
Thanks for the review @wli51 ! There will be a couple more prs shortly. Merging now
